### PR TITLE
build: add angular 17 support

### LINKED ISSUE #2078: 🚀[FEATURE]: Support Angular 17
### Relevant Package

All Ngxs packages.

### Description

Support Angular 17.

### Describe the problem you are trying to solve

Trying to update our codebase to Angular 17.

### Describe the solution you'd like

Relax the restriction in `package.json`.

### Describe alternatives you've considered

None at this time.

### PR DESCRIPTION
**Pull request by jpraetorius**
- extend allowed range of ng peer dependency to 18
- add an additional integration test for NG17 with updated dependencies and tests.
- add additional test in all places the other tests run

fixes #2078

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Updating an application using ngxs to Angular 17 fails, because all the packages declare `< 17` as the range for their peer dependency

Issue Number: #2078

## What is the new behavior?

Updating to Angular 17 works, the range of allowed Angular versions has been adapted to `<18`. Compatibility has been checked with an additional integration test.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Note that I first started by updating the dependencies for Angular to 17 too – that however results in a compile Error for the Angular 15 integration test, which is why I opted to not follow up on that change. The adaptation of `custom-rxjs-subject` below stems from that though (Updating Angular implies updating Typescript and the compiler complains about not being able to assert the type of `<T>` which made the explicit cast necessary.
I left that particular change in the commit, as I think it causes no harm and helps for further updates.

### CODE REVIEW

**Comment by jpraetorius** [comment]
For all I can gather about the error this seems to be related to old(ish) `yarn.lock` versions that might keep old browserslist configurations around. I don't know if you folks have a way of bypassing the cache for this build?
(I don't want to just update, as I get your reasoning why you check the file in and keep the cache for the build).

File: integrations/hello-world-ng17/src/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>HelloWorldNg16</title>
```
Comment (sdebeauchamp) [inline integrations/hello-world-ng17/src/index.html:5]: 16 -> 17

Comment (jpraetorius) [reply integrations/hello-world-ng17/src/index.html]: Thanks for spotting this one!

**Comment by monabbous** [comment]
Will this be reviewed soon?

@kuncevic @Carniatto @markwhitfeld @poloagustin @joaqcid @arturovt @splincode

**Comment by markwhitfeld** [comment]
Yes, my plan was to release today, but we have had a massive fibre break in
the area and I am without internet.
Hopefully tomorrow the release will be going out.

**Comment by ggros-septeo** [comment]
> Yes, my plan was to release today, but we have had a massive fibre break in the area and I am without internet. Hopefully tomorrow the release will be going out.

Thanks for you time!

**Review by markwhitfeld — APPROVED** [review]

**Comment by Sina7312** [comment]
When does this publish on npm?

**Comment by frasermclean** [comment]
Keen to add this to a new Angular 17 project

**Comment by CesarD** [comment]
Same here!! 😅

**Comment by danielbraghini** [comment]
any news about it?

**Comment by jpraetorius** [comment]
@danielbraghini as the timeline above shows this landed in 3.8.2

**Pull request merged**